### ISSUE TITLE
[Bug] A Node.js API is used (process.nextTick) which is not supported in the Edge Runtime

### ISSUE THREAD

**Issue by trevorpfiz** [pinecone-io/pinecone-ts-client#44]
### Is this a new bug?

- [X] I believe this is a new bug
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

`error - unhandledRejection: Failed to initialize Pinecone Client: PineconeClient: Error getting project name: Error: A Node.js API is used (process.nextTick) which is not supported in the Edge Runtime.`

import { PineconeClient } from '@pinecone-database/pinecone';

if (!process.env.PINECONE_ENVIRONMENT || !process.env.PINECONE_API_KEY) {
  throw new Error('Pinecone environment or api key vars missing');
}

Client:
```
async function initPinecone() {
  try {
    const pinecone = new PineconeClient();

    await pinecone.init({
      environment: process.env.PINECONE_ENVIRONMENT ?? '',
      apiKey: process.env.PINECONE_API_KEY ?? '',
    });

    return pinecone;
  } catch (error) {
    throw new Error(`Failed to initialize Pinecone Client: ${error.message}`);
  }
}

export const pinecone = await initPinecone();
```

Edge function:
```
 const filter = {
    metadataId: { $in: metadataIds },
  };

  const index = pinecone.Index(PINECONE_INDEX_NAME);
  const vectorStore = await PineconeStore.fromExistingIndex(
    new OpenAIEmbeddings({ modelName: 'text-embedding-ada-002' }),
    {
      namespace: userId,
      pineconeIndex: index,
      textKey: 'text',
      filter: filter,
    }
  );
```

### Expected Behavior

I would expect the Pinecone client to work in an edge function.

### Steps To Reproduce

1. Next.js edge function
2. attempt to use Pinecone client
3. see error

### Relevant log output

```shell
error - src/utils/pinecone.ts (18:10) @ initPinecone
error - unhandledRejection: Failed to initialize Pinecone Client: PineconeClient: Error getting project name: Error: A Node.js API is used (process.nextTick) which is not supported in the Edge Runtime.
```

### Environment

```markdown
- **OS**: Ubuntu 20.04.5 LTS with WSL2
- **Language version**: Node v18.15.0
- **Pinecone client version**: "@pinecone-database/pinecone": "^0.0.12"
```

### Additional Context

I have seen [lex-gpt](https://github.com/PineappleExpress808/lex-gpt/blob/07b05e8c36f5087a4614e0e716415825d25d81a1/pages/api/vectordbqa.ts#L19) attempting to use the client in an edge function. Was this working? I noticed it was ```"@pinecone-database/pinecone": "^0.0.10"```

**Comment by cm-igarashi-ryosuke** [comment]
I was able to get it to work with Edge Runtime by removing the following line
https://github.com/pinecone-io/pinecone-ts-client/blob/main/src/index.ts#L108

**Comment by francofantini** [comment]
I can confirm pinecone works when downgrading to `v0.0.10`. Downgrading for now.

Any news on a fix?

**Comment by ElektrikSpark** [comment]
Can confirm `v0.0.10` works, but `v0.0.14` does not due to use of `process.nextTick`.

**Comment by pirtlj** [comment]
Ran into this today when trying to run pinecone with Langchain on Vercel / Nextjs  Edge

**Comment by jhamon** [comment]
Just published `v0.1.5` which no longer contains the problematic `nextTick` call. Going to close this, but please let us know if that does not resolve the issue.
